.danger/utils.ts

```diff
@@ -0,0 +1,94 @@
+import {PackageShape} from './types';
+
+/**
+ * Returns a list of public packages by filtering out the private ones
+ *
+ * @param {PackageShape[]} packageList
+ */
+export const getPublicPackages = (packageList: PackageShape[]) => packageList.filter((p) => p.private === false);
+
+/**
+ * Returns a file path relative to the root of the project from a full machine file path
+ * by only keeping the last part of the path including `packages`
+ *
+ * @param {string} location
+ * @return {*}  {string}
+ */
+export const getRepoPackagePath = (location: string): string =>
+  location.slice(location.indexOf('packages'), location.length);
+
+/**
+ * Returns file paths relative to the root of the project from a list of packages
+ *
+ * @param {PackageShape[]} packages
+ * @return {*}  {string[]}
+ */
+export const getPackagePaths = (packages: PackageShape[]): string[] =>
+  packages.map(({location}) => getRepoPackagePath(location));
+
+/**
+ *  From a list of files, return only the files that are located in
+ * the folders of public packages
+ *
+ * @param {string[]} fileList
+ * @param {string[]} publicPackagePaths
+ */
+export const getPubicPackageFilesFromFiles = (fileList: string[], publicPackagePaths: string[]) =>
+  fileList.filter((filePath) => {
+    return publicPackagePaths.filter((path) => filePath.includes(path)).length > 0;
+  });
+
+/**
+ * Return the name of the package a file belongs to
+ *
+ * @param {string} filePath
+ * @param {PackageShape[]} publicPackages
+ * @return {*}  {string}
+ */
+export const getPackageNameFromPath = (filePath: string, publicPackages: PackageShape[]): string => {
+  const packageNames: string[] = [];
+  publicPackages.forEach((p) => {
+    if (filePath.includes(`${getRepoPackagePath(p.location)}/`)) {
+      packageNames.push(p.name);
+    }
+  });
+  return packageNames[0];
+};
+
+/**
+ * From a list of files return the names of the packages that would need to be published
+ * to npm based on what those files were.
+ *
+ * @param {string[]} touchedFiles
+ * @param {PackageShape[]} publicPackages
+ * @return {*}  {string[]}
+ */
+export const getUnpublishedPackageNames = (touchedFiles: string[], publicPackages: PackageShape[]): string[] => {
+  const uniquePackages: string[] = [];
```

Comment:
nit: If we make this a set, we won't have to check for uniqueness since Sets are unique in nature:

```js
const uniquePackages = new Set();

touchedFiles.forEach((filePath) => {
  const packageName = getPackageNameFromPath(filePath, publicPackages);

  if (filePath.includes('/src/')) {
    uniquePackages.add(packageName);
  }
});

// Convert the set back to an array, if you need it as an array
return [...uniquePackages];
```

Reply:
I like